components/background_hang_monitor/sampler_mac.rs

```diff
@@ -104,6 +105,11 @@ unsafe fn frame_pointer_stack_walk(regs: Registers) -> NativeStack {
     let mut current = regs.frame_ptr as *mut *mut std::ffi::c_void;
     while !current.is_null() {
         if (current as usize) < stackaddr as usize {
+            // Reached the end of the stack.
+            break;
+        }
+        if (usize::max_value() / current as usize) <= 1 {
```

Comment:
We can call [pthread_get_stacksize_np](https://docs.rs/libc/0.2.48/i686-apple-darwin/libc/fn.pthread_get_stacksize_np.html) to determine the other end of the stack. That seems like it will be more reliable than this heuristic.

Reply:
I replaced this with `if current as usize >= stackaddr.add(stacksize * 8) as usize`, with the thinking behind `* 8` being 64 bit mac... Does that make sense?